plugins/repository-s3/src/main/java/org/elasticsearch/repositories/s3/S3Repository.java

```diff
@@ -58,149 +58,67 @@
      * NOTE: These are legacy settings. Use the named client config settings above.
      */
     public interface Repositories {
```

Comment:
Not part of this change but you can remove `public` here. Also for `public static final String TYPE = "s3";`